docs: add blog for data/infra/tools picks of 2022 (#272)
bytebase/star-history: public/blog/star-history-yearly-pick-2022-data-infra-tools.md

## Bytebase

![bytebase](/blog/assets/yearly-pick-data-infra-tools-2022/bytebase.webp)

While the aforementioned Buf targets the API schema, [Bytebase](https://github.com/bytebase/bytebase) targets the database schema hidden behind the API. While there are a number of commercial/open-source database engines on the market, there are only a few tools around to ensure reliable database schema changes. To this day, most engineering teams continue to connect directly to production databases to change data. 💥

- Like how GitHub/GitLab manages code changes, Bytebase provides a GUI to manage database changes.
- Like Figma for designer team collaboration, Bytebase helps DevOps teams collaborate on database management.
- Like  how Terraform defines Infrastructure-as-Code, Bytebase aims to build Database-as-Code.

![bytebase-site](/blog/assets/yearly-pick-data-infra-tools-2022/bytebase-site.webp)

Bytebase is a Database CI/CD tool (Database-as-Code) for DevOps teams, built for developers and DBAs, and the first (and at the time of this post, only) [Database CI/CD solution to be included in the CNCF Landscape](https://www.bytebase.com/blog/cncf-landscape).

> The mascots Byte and Base (originated from Bytebase's B, and the shape of a database, respectively) are also adorable.

Reviewer: gif missing?

*Note: the list is in alphabetical order.

## Buf

![buf](/blog/assets/yearly-pick-data-infra-tools-2022/buf.webp)

[Buf](https://github.com/bufbuild/buf) is a very low-profile company based in Toronto that focuses on improving the developer experience with Protocol Buffers. Their mission is to deprecate REST/JSON and move the industry towards a schema-driven development model with Protobuf. Their products the Buf CLI and Buf Schema Registry help engineers shift most of their time and effort from managing Protobuf files to implementing features and working on their infrastructure.

Buf closed a $68M Series B at the end of 2021 and has kept a low profile for all four rounds of funding, with [the only official statement](https://buf.build/blog/an-update-on-our-fundraising) says that they want people to focus on the API and Protobuf ecosystem, rather than how the company is financing the journey (alas, funding is still a big part of the journey).

![buf-site](/blog/assets/yearly-pick-data-infra-tools-2022/buf-site.webp)

## Bytebase

![bytebase](/blog/assets/yearly-pick-data-infra-tools-2022/bytebase.webp)

> While the aforementioned Buf targets the API schema, [Bytebase](https://github.com/bytebase/bytebase) targets the database schema hidden behind the API. While there are a number of commercial/open-source database engines on the market, there are only a few tools around to ensure reliable database schema changes. To this day, most engineering teams continue to connect directly to production databases to change data. 💥

Reviewer 2: ```suggestion
While the aforementioned Buf targets the API schema, [Bytebase](https://github.com/bytebase/bytebase) targets the database schema hidden behind the API. While there are a number of commercial/open-source database engines on the market, there are only a few tools around to ensure reliable database schema changes. To this day, many engineering teams continue to connect directly to production databases to change data. 💥
```

![steampipe-hn](/blog/assets/yearly-pick-data-infra-tools-2022/steampipe-hn.webp)

## Tier

![tier](/blog/assets/yearly-pick-data-infra-tools-2022/tier.webp)

[Tier](https://github.com/tierrun/tier) is a tool to manage pricing and billing workflow for your SaaS offering

Tier landed on [Hacker News front page](https://news.ycombinator.com/item?id=33429972), and their stars skyrocketed, just like Steampipe.

![tier-hn](/blog/assets/yearly-pick-data-infra-tools-2022/tier-hn.webp)

Pricing SaaS products is a constant WIP, but it is a complex task and is often tied up in complex workflows and hidden cross-organizational dependencies. Many startups choose a pricing strategy for their product once, which stays the same forever.

Tier has a top-level design for how to implement pricing that allows for changes to pricing models without changing application code or business processes (Pricing as Code). The founding team thoroughly thought about pricing and built a website explaining t[he 5 Pillars of PriceOps](https://priceops.org/).

![tier-priceops](/blog/assets/yearly-pick-data-infra-tools-2022/tier-priceops.webp)

> This time around, Star History is highlighting some relatively lesser-known projects that are on the rise. We see that the **Data & Infra & Tools** section can be divided into two main categories:

Reviewer 2: ```suggestion
For Star History, our goal is to discover interesting and inspiring projects. The aforementioned projects are not household names in the open source community (yet), but each has great potential since they are tackling genuine issues with novel ideas. We see that the **Data & Infra & DevTools** section can be divided into two main categories:
```

> - Vertical innovation: database engines being the most popular direction, such as DuckDB and Neon.

Reviewer 2: ```suggestion
- Vertical innovation: database engines being the most popular segment, such as DuckDB and Neon.
```

- Horizontal unification and standardization:
  - Buf for API development.
  - Bytebase for database development.
  - Casdoor for authentication sources.
  - Dagger for pipeline deployments.

>   - Meltano for large database platforms.

Reviewer 2: ```suggestion
  - Meltano for data platforms.
```

  - Opa for policy configurations.

>   - Steampipe for accessing cloud resources under different cloud providers.

Reviewer 2: ```suggestion
  - Steampipe for accessing cloud resources across different cloud providers.
```
